### ISSUE TITLE
keystrokes entered while DPMS is off aren't recognized properly.

### ISSUE THREAD

**Issue by crocket** [swaywm/swaylock#188]
This doesn't happen on my desktop, but it happens on my laptop.

My laptop doesn't have elogind. My desktop computer does. Perhaps, it is related to elogind?

How can I debug this issue?

**Comment by mindrunner** [comment]
Same boat here. Not sure if I am using `seatd` or `elogind`. Using `sddm` as a display manager.

**Comment by crocket** [comment]
```
[unix_chkpwd] password check failed for user (xxx)
[swaylock] pam_unix(swaylock:auth): authentication failure; logname= uid=1000 euid=1000 tty= ruser= rhost=  user=xxx
```

**Comment by kushraj** [comment]
happening without any display manager too with systemd

**Comment by crocket** [comment]
elogind is a part of systemd.

**Comment by emersion** [comment]
That is incorrect, elogind is a fork of systemd with everything removed except logind.

**Comment by crocket** [comment]
I mean elogind is almost the same as logind from systemd.

**Comment by kennylevinsen** [comment]
I very much doubt this would have anything to do with seat management of any kind.

I'm more inclined to think it might be a focus handling corner case when outputs are off/re-enable. Or maybe lost input events as sway stalls during output configuration.

**Comment by adee** [comment]
The cause is the input reconfigure during DPMS on.
You can try my patch from here: https://github.com/swaywm/sway/issues/5757#issuecomment-882028355

**Comment by kushraj** [comment]
@adee pastebin link doesnt work anymore, can you please share a gist instead

**Comment by pielgrzym** [comment]
I have a similar problem - when I lock with swayidle I make dpms kick in immediately and turn off the monitor. When I start typing when screen is still off I  won't be able to log in. Here is the twist: I just typed *one letter* while screen was off. It turned on the screen and displayed swaylock. When I was pressing backspace I notice it actually typed 4 letters - this is why the password was wrong on first try.
Can anyone confirm this behavior?

**Comment by mindrunner** [comment]
Exatly this!!

Workaround:

Type `<ESC>[password]<ENTER>` to unlock

**Comment by pielgrzym** [comment]
Yeah, I'm getting used to hitting 'Ctrl' before I start typing :rofl:

**Comment by hramrach** [comment]
Swaylock could somewhat alleviate this by indicating how many characters are typed, eg. making the width of the colored part of the ring that's shown in response to keypress fraction of the circle proportional to number of characters typed rather than fixed.

**Comment by kennylevinsen** [comment]
> Swaylock could somewhat alleviate this by indicating how many characters are typed

The whole point of swaylock and i3lock's randomized circular typing animation is to *not* reveal how many characters have been typed at any point, so that someone looking over your should won't easily be able to see your password length.

**Comment by hramrach** [comment]
And with the proposed change they still won't so long as your password is not 5 characters long. The point is to make the initial few characters easy to distinguish so the error of having unexpectedly many at the start is avoided but not make the number typed so far obvious once more than a few are typed.

**Comment by laktak** [comment]
> The whole point of swaylock and i3lock's randomized circular typing animation is to not reveal how many characters have been typed at any point,

@kennylevinsen it could show that the buffer is empty though

also see #132
